vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/api/FeedClientImpl.java

```diff
@@ -45,17 +45,12 @@ public FeedClientImpl(SessionParams sessionParams,
     }
 
     @Override
-    public void stream(String documentId, CharSequence documentData) {
-        stream(documentId, documentData, null);
-    }
-
-    @Override
-    public void stream(String documentId, CharSequence documentData, Object context) {
+    public void stream(String documentId, String operationId, CharSequence documentData, Object context) {
         CharsetEncoder charsetEncoder = StandardCharsets.UTF_8.newEncoder();
         charsetEncoder.onMalformedInput(CodingErrorAction.REPORT);
         charsetEncoder.onUnmappableCharacter(CodingErrorAction.REPORT);
 
-        Document document = new Document(documentId, documentData, context);
+        Document document = new Document(documentId, operationId, documentData, context);
```

Comment:
We don't want to specify the same operation id for multiple documents!
The problem is that the sender may need to know about the operation id before receiving the callback for the operation (to determine if it issued the operation), and if the operation id is not known before after issuing the operation we enter a race between the callback and the sender preparing to receive it. 

The simple way to avoid this is to let the sender decide it ... with the drawback that we introduce a new shiny way for it to shoot self in foot. I'll add some doc on this.

Reply:
Got it, was a bit confused as to the relationship between an operation and a document ID in this context.